src/providers/profile/profile.ts

```diff
@@ -1438,60 +1426,83 @@ export class ProfileProvider {
     return new Promise((resolve, reject) => {
+
+      if (tokens && tokens.length && coins.indexOf('eth') <= 0) {
+        reject('No ethereum wallets for tokens');
+      }
+
       const defaultOpts = this.getDefaultWalletOpts(coins[0]);
+
       this._createWallet(defaultOpts)
         .then(data => {
           const key = data.key;
-          const firstWalletClient = data.walletClient;
-
-          // Encrypt wallet
-          this.onGoingProcessProvider.pause();
-          this.askToEncryptKey(key).then(password => {
-            this.onGoingProcessProvider.resume();
-            this.keyProvider.addKey(key).then(() => {
-              const promises = [];
-              coins.slice(1).forEach(coin => {
-                const newOpts: any = {};
-                Object.assign(newOpts, this.getDefaultWalletOpts(coin));
-                newOpts['keyId'] = key.id; // Add Key
-                if (password) newOpts['password'] = password;
-                promises.push(this._createWallet(newOpts));
-              });
-              Promise.all(promises)
-                .then(wallets => {
-                  wallets.unshift({ walletClient: firstWalletClient, key });
-                  const bindWalletClients = [];
-                  wallets.forEach(w => {
-                    bindWalletClients.push(
-                      this.addAndBindWalletClient(w.walletClient, {
-                        bwsurl: defaultOpts.bwsurl
-                      })
-                    );
-                  });
-                  this.storeProfileIfDirty().then(() => {
-                    Promise.all(bindWalletClients)
-                      .then(walletClients => {
-                        this.events.publish('Local/WalletListChange');
-                        return resolve(walletClients);
-                      })
-                      .catch(e => {
-                        reject(e);
-                      });
-                  });
+          const firstWalletData = data;
+
+          this.keyProvider.addKey(key).then(() => {
+            const create2ndWallets = [];
+            coins.slice(1).forEach(coin => {
+              const newOpts: any = {};
+              Object.assign(newOpts, this.getDefaultWalletOpts(coin));
+              newOpts['keyId'] = key.id; // Add Key
+              create2ndWallets.push(this._createWallet(newOpts));
+            });
+            Promise.all(create2ndWallets)
+              .then(datas => {
+                datas.unshift(firstWalletData);
+                let walletClients = _.map(datas,'walletClient');
+
+                // Handle tokens
+                const ethWalletClient = walletClients.find(wallet => wallet.credentials.coin === 'eth');
+                let tokenCreate  = [];
+                if (ethWalletClient) {
+                  tokenCreate = tokens.map(token => this.createTokenWallet(ethWalletClient, token));
+                }
+                Promise.all(tokenCreate).then( tokenClients => {
```

Comment:
Since `createTokenWallet` is not an async function, I think that "Promise.all" is not necessary

Reply:
thanks! fixing.